repair + compact: Improved logging for easier future debug purposes.

Bplotka opened the pull request:
This is based on my experience while debugging https://github.com/prometheus/prometheus/issues/3943.

I needed to deduct few things, and all that would be just bit easier with these two logs:
- new block's ULID on each compaction.
- actual list of Blocks (ulid + time range) on Prometheus startup (easy to log that while repairing blocks).

We don't really need blocks that takes part in compaction - that can be deducted easily based on time ranges of blocks we have currently in system.

Additionally, fixed comments.

What do you think @fabxc  @Gouthamve ?

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

Bplotka commented on compact.go:
```diff
@@ -482,6 +480,14 @@ func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockRe
 		return errors.Wrap(err, "rename block dir")
 	}
+
+	level.Info(c.logger).Log(
```
IMO we need to log on successful compaction only, right?

Gouthamve replied:
Hmm, if it fails, it is good to know which blocks were involved. Is it worth printing out the blocks being compacted as well? And print out before the compaction?

Bplotka replied:
Yea, my worry is that we log this and assumed everything is fine and the error is for some other compaction.. what do you think?

Shouldn't we make sure our `errors.Wrap` have all required info (what blocks are involved) in case of error?

I will end up passing here compacted block metas as well for this, but maybe that's ok.

Gouthamve replied:
Fair enough, yep :) Need to fix our errors.

fabxc commented on compact.go:
```diff
@@ -482,6 +480,14 @@ func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockRe
+		"count", len(blocks),
+		"mint", meta.MinTime,
+		"maxt", meta.MaxTime,
+		"ulid", meta.ULID.String(),
```
You can leave out the `.String()` I think.

fabxc commented on compact.go:
```diff
@@ -482,6 +480,14 @@ func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockRe
+
+	level.Info(c.logger).Log(
+		"msg", "compact blocks",
+		"count", len(blocks),
```
Maybe we should also just log all input blocks. A bit noisy but at least we can deduce the exact history in debugging cases.

Bplotka replied:
As mentioned in the description - I can deduct that easily given that I know what is the state on startup. But I guess  there is no harm in printing these here too.

fabxc replied:
Yea, seems like a lot of mental work though compared to just having it printed right there.

Bplotka commented:
@Gouthamve  @fabxc Addressed comments. 

@Gouthamve I double checked and our wrap for `Compact` already include `planned`  dirs so they will be printed in case of error (:

Gouthamve commented:
```
level=info msg="write block" mint=1518912000000 maxt=1518919200000 ulid=01C8N2CB9RTXYTZ9JTWNK8DAD6
level=info msg="head GC completed" duration=1.34756ms
level=info msg="WAL truncation completed" duration=960ns
.
.
.
level=info msg="write block" mint=1518933600000 maxt=1518940800000 ulid=01C8N2CDSHRNX0C32QZTGFMSPM
level=info msg="head GC completed" duration=774.55µs
level=info msg="WAL truncation completed" duration=831ns
level=info msg="compact blocks" count=3 mint=1518912000000 maxt=1518933600000 ulid=01C8N2CDXQ77D06H5WGV3KD3G1 sources="unsupported value type"
```

Arrays are not automatically printed I guess.

Bplotka commented:
thanks for checking ): Was too sure about it!

Bplotka commented:
@Gouthamve  Sorry for that - I fixed it and tested log line manually.

Gouthamve commented:
np, Works perfectly now! :+1: :+1: :+1:
